Refactor timestamp formatting in Git UI components to use `chrono` for local time calculations

virajbhartiya opened the pull request:
- Updated `blame_ui.rs`, `branch_picker.rs`, `commit_tooltip.rs`, and `commit_view.rs` to replace the previous timestamp formatting with `chrono` for better accuracy in local time representation.
- Introduced `chrono::Local::now().offset().local_minus_utc()` to obtain the local offset for timestamp formatting.

Closes #40878

Release Notes:
- Improved timestamp handling in various Git UI components for enhanced user experience.

virajbhartiya committed:
Refactor timestamp formatting in Git UI components to use `chrono` for local time calculations

- Updated `blame_ui.rs`, `branch_picker.rs`, `commit_tooltip.rs`, and `commit_view.rs` to replace the previous timestamp formatting with `chrono` for better accuracy in local time representation.
- Introduced `chrono::Local::now().offset().local_minus_utc()` to obtain the local offset for timestamp formatting.

Release Notes:
- Improved timestamp handling in various Git UI components for enhanced user experience.

virajbhartiya committed:
Remove unused `format_local_timestamp` imports from Git UI components

- Cleaned up `blame_ui.rs`, `branch_picker.rs`, and `commit_tooltip.rs` by removing the `format_local_timestamp` import, which is no longer needed after the recent refactor to use `chrono` for timestamp formatting.

Release Notes:
- Code cleanup in Git UI components to enhance maintainability.

cole-miller reviewed (changes requested):
Thanks! A couple of small bits of feedback.

virajbhartiya committed:
Update dependencies and refactor timestamp handling in Git UI components

- Added the "local-offset" feature to the `time` crate in `Cargo.toml`.
- Removed the unused `chrono` import from `branch_picker.rs`.
- Replaced `chrono::Local::now().offset().local_minus_utc()` with `UtcOffset::current_local_offset().unwrap_or(UtcOffset::UTC)` in `commit_tooltip.rs` for local time calculations.

virajbhartiya commented:
@cole-miller I've addressed the changes mentioned

virajbhartiya committed:
Remove unused imports

cole-miller commented on crates/git_ui/src/blame_ui.rs:
```diff
@@ -188,9 +188,11 @@ impl BlameRenderer for GitBlameRenderer {
             .map(|sha| sha.to_string().into())
             .unwrap_or_else(|| sha.clone());
-        let absolute_timestamp = format_local_timestamp(
+        let local = chrono::Local::now().offset().local_minus_utc();
```
This is still using chrono

cole-miller commented on crates/git_ui/src/commit_view.rs:
```diff
@@ -415,12 +416,14 @@ fn format_commit(commit: &CommitDetails, is_stash: bool) -> String {
         commit.author_name, commit.author_email
     )
     .unwrap();
+    let local = chrono::Local::now().offset().local_minus_utc();
```
This is still using chrono.

cole-miller commented on crates/git_ui/src/branch_picker.rs:
```diff
@@ -447,9 +446,11 @@ impl PickerDelegate for BranchListDelegate {
                 let commit_time = OffsetDateTime::from_unix_timestamp(commit.commit_timestamp)
                     .unwrap_or_else(|_| OffsetDateTime::now_utc());
-                let formatted_time = format_local_timestamp(
+                let local = chrono::Local::now().offset().local_minus_utc();
```
This is still using chrono

cole-miller reviewed (changes requested):
It looks like there are still some uses of chrono in the latest revision, could you go through and update these as discussed in the previous review comments?

virajbhartiya committed:
Refactor timestamp handling in Git UI components to use `time` crate

- Removed unused `chrono` imports from `blame_ui.rs`, `branch_picker.rs`, `commit_view.rs`, and `stash_picker.rs`.
- Replaced `chrono::Local::now().offset().local_minus_utc()` with `UtcOffset::current_local_offset().unwrap_or(UtcOffset::UTC)` for local time calculations across multiple files.

virajbhartiya committed:
Remove unused `chrono` dependency from `Cargo.toml` and `Cargo.lock`

- Deleted `chrono` from the dependencies in `Cargo.toml` and `Cargo.lock` as it is no longer needed.
- Updated timestamp handling in `blame_ui.rs` and `branch_picker.rs` to improve clarity by formatting the local offset calculation across multiple lines.

cole-miller reviewed (approved):
Thanks!
